src/clients/HubPoolClient.ts

```diff
@@ -650,11 +647,60 @@ export class HubPoolClient {
     this.disputedRootBundles.push(
       ...events["RootBundleDisputed"].map((event) => spreadEventWithBlockNumber(event) as DisputedRootBundle)
     );
-    this.executedRootBundles.push(
-      ...events["RootBundleExecuted"]
-        .filter((event) => !this.configOverride.ignoredHubExecutedBundles.includes(event.blockNumber))
-        .map((event) => spreadEventWithBlockNumber(event) as ExecutedRootBundle)
-    );
+
+    const configStoreVersions: { [blockNumber: number]: number } = {};
+    for (const event of events["RootBundleExecuted"]) {
+      if (this.configOverride.ignoredHubExecutedBundles.includes(event.blockNumber)) {
+        continue;
+      }
+
+      // The applicable version is determined by the block number of the corresponding proposal.
```

Comment:
Some thoughts on this:
 - The HubPool does not permit new proposals whilst there are existing un-executed leaves, so it effectively enforces that the proposal immediately before this executed leaf is the right one.
 - We assume that the RPC provider supplies events chronologically and completely.

I'd feel better if we verified the link between the root bundle proposal and the executed leaf with something stronger than the relationship between their blocks, but it's quite difficult to do that this low in the stack. Trying to do so may introduce chicken and egg problems.